Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+* try to connect to the iot hub to make sure that the passed in values are valid!
+*/
+function setConfigs(req, res)
+{
+  DEVICE_ID=req.body["device-id"];
+  IOTHUB_CONNECTION_STRING=req.body["iothub-connection-string"];
+  convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+  {
+    IOTHUB_ENDPOINT = eventhubString;
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send(error.message);
+  });
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) => 
+{
+    wss.clients.forEach((client) => 
+    {
+      if (client.readyState === WebSocket.OPEN) 
+      {
+        try 
+        {
+          client.send(data);
+        } catch (er) 
+        {
+          console.error(er);
+        }
+      }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+*/
+async function receiveHubMessages() 
+  {
+      const consumerGroup = "$Default"; // name of the default consumer group
+      eventHubReader = new EventHubReader(IOTHUB_CONNECTION_STRING, consumerGroup);
+      (async () => 
+      {
+          await eventHubReader.startReadMessage((message, date, deviceId) => 
+          {
+            try 
+            {
+              const payload = 
+              {
+                IotData: message,
+                MessageDate: date || Date.now().toISOString(),
+                DeviceId: deviceId,
+              };
+
+              wss.broadcast(JSON.stringify(payload));
+            } catch (err) 
+            {
+              console.error('Error broadcasting: [%s] from [%s].', err, message);
+            }
+
+          });
+      })().catch();
+}
+
+/**
+ * Connects to IoT Hub and returns a list of currently running modules 
+ * @returns {Promise<ResultWithHttpResponse<Module[]>>}
+ */
+function iotHubConnection(req, res)
+{
+    let registry = iothub.Registry.fromConnectionString(IOTHUB_CONNECTION_STRING);
+    return registry.getModulesOnDevice(DEVICE_ID);
+};
+
+/**
+* checks that the iothub_connection_string and device ID are set. If not, responds to client wth 404 and error message
+* @param {XMLHttpRequestResponseType} res - request response
+* @returns {boolean} - true if credential are set
+*/
+function validCredentialsAreSet(res)
+{
+  if (IOTHUB_CONNECTION_STRING == "" || DEVICE_ID == "")
+  {
+    res.status(404).send("Please set your IoT Hub Connection String and Device ID!");
+    return false;
+  }
+  return true;
+}
+
+/**
+* this function invokes a direct method on the lvaEdge module.
+ * @returns {Promise<ResultWithHttpResponse<any>>}
+*/
+function invokeLVAMethod(req, res)
+{
+    let client = Client.fromConnectionString(IOTHUB_CONNECTION_STRING);
+
+    return client.invokeDeviceMethod(DEVICE_ID, MODULE_ID,
+        {
+            methodName: req.body.methodName,
+            payload: req.body.Payload,
+            responseTimeoutInSeconds: 200,
```

Comment:
any particular reason for these timeout values? what happens if they're not set (aka what are the defaults)?

Reply:
Can omit. Default is 200